target_api/sinks.py

```diff
@@ -71,3 +75,34 @@ def upsert_record(self, record: dict, context: dict):
             self.logger.warning(f"Unable to get response's id: {e}")
 
         return id, response.ok, dict()
+    
+    @backoff.on_exception(
+    backoff.expo,
+    (RetriableAPIError, requests.exceptions.ReadTimeout),
+    max_tries=10,
+    factor=2,
+    )
+    def _request(
+        self, http_method, endpoint, params={}, request_data=None, headers={}
+    ) -> requests.PreparedRequest:
+        """Prepare a request object."""
+        url = self.url(endpoint)
+        headers.update(self.default_headers)
+        headers.update({"Content-Type": "application/json"})
+        params.update(self.params)
+        data = (
+            json.dumps(request_data, cls=HGJSONEncoder)
+            if request_data
+            else None
+        )
+
+        response = requests.request(
+            method=http_method,
+            url=url,
+            params=params,
+            headers=headers,
+            data=data,
+            auth = ("PRECORO", "Sage2023!")
```

Comment:
Why is this hard coded??

Reply:
I used for testing and forgot to remove, it's removed from the PR now.